src/RootSolvers.jl

```diff
@@ -538,40 +1053,57 @@ function find_zero(
     tol::AbstractTolerance,
     maxiters::Int,
 ) where {F <: Function, FT}
-    x_history = init_history(soltype, FT)
-    y_history = init_history(soltype, FT)
-    if soltype isa VerboseSolution
-        y, y′ = f(x0)
-        push_history!(x_history, x0, soltype)
-        push_history!(y_history, y, soltype)
-    end
-    for i in 1:maxiters
-        y, y′ = f(x0)
-        x1 = x0 - y / y′
-        push_history!(x_history, x1, soltype)
-        push_history!(y_history, y, soltype)
-        if tol(x0, x1, y)
-            return SolutionResults(
-                soltype,
-                x1,
-                true,
-                y,
-                i,
-                x_history,
-                y_history,
-            )
-        end
-        x0 = x1
-    end
-    return SolutionResults(
-        soltype,
-        x0,
-        false,
-        y,
-        maxiters,
-        x_history,
-        y_history,
-    )
+    return _find_zero_newton(f, x -> f(x)[1], x0, soltype, tol, maxiters)
+end
+
+"""
+    method_args(method::RootSolvingMethod)
+
+Extract the intial guess(es) for a root-solving method for internal dispatch.
+
+This function is used internally to unpack method parameters for passing to the
+appropriate `find_zero` implementation.
+
+## Arguments
+- `method::RootSolvingMethod`: The root-solving method instance
+
+## Returns
+- `Tuple`: Initial guess(es) specific to the method type
+
+## Example
+```julia
+method = SecantMethod{Float64}(0.0, 1.0)
+args = method_args(method)  # Returns (0.0, 1.0)
+```
+"""
```

Comment:
This and other methods that are in the "Internal methods" section of the API docs would probably be useful to include in/move to a developer docs section.

Reply:
Done.